pkg/cli/admin/release/bug.go

```diff
@@ -9,61 +9,61 @@ import (
 	"net/url"
 	"regexp"
 	"sort"
-	"strconv"
 	"strings"
-
-	"k8s.io/klog/v2"
 )
 
-type BugType int
+type RefType int
 
 const (
-	Bugzilla BugType = iota
+	Bugzilla RefType = iota
 	Jira
 )
 
 const (
 	maxRetries = 2
 
-	bugzillaBrowseURL = "https://bugzilla.redhat.com/show_bug.cgi?id=%d"
+	bugzillaBrowseURL = "https://bugzilla.redhat.com/show_bug.cgi?id=%s"
 	bugzillaRestURL   = "https://bugzilla.redhat.com/rest/bug"
-	jiraBrowseURL     = "https://issues.redhat.com/browse/OCPBUGS-%d"
-	JiraRestURL       = "https://issues.redhat.com/rest/api/latest/issue/OCPBUGS-%d"
+	jiraBrowseURL     = "https://issues.redhat.com/browse/%s"
+	JiraRestURL       = "https://issues.redhat.com/rest/api/latest/issue/%s"
 )
 
 var (
+	// find strings like "Bug: 123,456 789: some commit description"
 	reBugzillaBug = regexp.MustCompile(`^[B|b]ug(?:s)?\s([\d\s,]+)(?:-|:)\s*`)
-	reJiraBug     = regexp.MustCompile(`^OCPBUGS-([\d\s,]+)(?:-|:)\s*`)
+
+	// find strings like "XYZ-123,ABC-456 DEF-789: some commit description"
+	reJiraRef = regexp.MustCompile(`^([A-Z]+-\d+[\s,]*)+(?:-|:)\s*`)
```

Comment:
I'd like to support the existing `NO-ISSUE` semantic that some [teams are using](https://amd64.ocp.releases.ci.openshift.org/releasestream/4-dev-preview/release/4.13.0-ec.1).  If we do enforce, I think we might want a standard escape hatch.

Reply:
this code doesn't enforce anything, this is just for adding links to the changelog so they can show up in the release controller payload diff report (and so we can potentially use those links to scrape more information about what went into the payload, based on the jira ticket that is linked).  

Since "no-issue" items by definition have nothing to link to, they should be ignored by this code (which they will be, since no-issue/no-jira won't match the regex)

the enforcement logic would be done in the prow bot + tide merge label requirements, which is unrelated to this codebase (other than in that both of them consume the same text: the pr title, effectively)

but yes, when we get to the prow bot logic, we'll want it to respect "no-issue/no-jira" as a valid "jira reference" for purposes of applying the "valid-jira" label to a PR.